config/clients/dotnet/template/Telemetry/Attributes.cs.mustache

```diff
@@ -0,0 +1,227 @@
+        }
+
+        return null;
+    }
+
+    /**
+     * Builds an object of attributes that can be used to report alongside an OpenTelemetry metric event.
+     *
+     * @param response - The Axios response object, used to add data like HTTP status, host, method, and headers.
+     * @param credentials - The credentials object, used to add data like the ClientID when using ClientCredentials.
+     * @param methodAttributes - Extra attributes that the method (i.e. check, listObjects) wishes to have included. Any custom attributes should use the common names.
+     * @returns {Attributes}
+     */
+    public static KeyValuePair<string, object>[] buildAttributesForResponse(string apiName,
+        HttpResponseMessage response, RequestBuilder requestBuilder, Credentials? credentials,
+        Stopwatch requestDuration, int retryCount) {
+        var attributes = new List<KeyValuePair<string, object>>();
+
+        // To match the JS SDK, we are converting the method name to camelCase
+        attributes.Add(new KeyValuePair<string, object>(AttributeRequestMethod, apiName));
+
+        if (requestBuilder.PathParameters.ContainsKey("store_id")) {
+            var storeId = requestBuilder.PathParameters.GetValueOrDefault("store_id");
+            if (!string.IsNullOrEmpty(storeId)) {
+                attributes.Add(new KeyValuePair<string, object>(AttributeRequestStoreId, storeId));
+            }
+        }
+
+        string? modelId = null;
+        // if the model id is in the path params try to get it from there
+        if (requestBuilder.PathParameters.ContainsKey("authorization_model_id")) {
+            modelId = requestBuilder.PathParameters.GetValueOrDefault("authorization_model_id");
+            if (!string.IsNullOrEmpty(modelId)) {
+                attributes.Add(new KeyValuePair<string, object>(AttributeRequestModelId, modelId));
+            }
+        }
+        // In the case of ReadAuthorizationModel, the path param is called ID
+        else if (requestBuilder.PathTemplate == "/stores/{store_id}/authorization-models/{id}" &&
+                 requestBuilder.PathParameters.ContainsKey("id")) {
+            modelId = requestBuilder.PathParameters.GetValueOrDefault("id");
+            if (!string.IsNullOrEmpty(modelId)) {
+                attributes.Add(new KeyValuePair<string, object>(AttributeRequestModelId, modelId));
+            }
+        }
+        // In many endpoints authorization_model_id is sent as a field in the body
+        // if the apiName is Check or ListObjects, we always want to parse the request body to get the model ID and the user (subject)
+        // if the apiName is Write, Expand or ListUsers we want to parse it to get the model ID
+        else if (apiName is "Check" or "ListObjects" or "Write" or "Expand" or "ListUsers") {
+            try {
+                var partialApiResquest =
+                    JsonSerializer.Deserialize<IPartialApiRequest>(requestBuilder.Body?.ToString()!);
```

Comment:
`HttpContent` will need its stream deserialized, `ToString` isn't overridden.

Reply:
Does `JsonDocument` offer an optimization over `JsonNode`? Would you recommend converting to `JsonDocument` instead?

Reply:
Yes, `JsonDocument` is optimized for immutable reads. Just make sure to dispose it as it uses pooled memory.

Reply:
Thanks for the tip, we've done that change